packages/horizontal-slider-js/src/horizontalSlider.tsx

```diff
@@ -0,0 +1,68 @@
+/** @jsx h */
+import {
+  createHorizontalSliderComponent,
+  generateHorizontalSliderId,
+  HorizontalSliderProps,
+  RecordWithObjectID,
+  updateNavigationButtonsProps,
+} from '@algolia/ui-components-horizontal-slider-vdom';
+import { createElement, Fragment, h, render } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+
+import { getHTMLElement } from './getHTMLElement';
+import { EnvironmentProps } from './types';
+
+const UncontrolledHorizontalSlider = createHorizontalSliderComponent({
+  createElement,
+  Fragment,
+});
+
+function HorizontalSlider<TObject extends RecordWithObjectID>(
+  props: HorizontalSliderProps<TObject>
+) {
+  const listRef = useRef<HTMLOListElement>(null);
+  const nextButtonRef = useRef<HTMLButtonElement>(null);
+  const previousButtonRef = useRef<HTMLButtonElement>(null);
+  const sliderIdRef = useRef(generateHorizontalSliderId());
+
+  useEffect(() => {
+    updateNavigationButtonsProps({
+      listRef,
+      nextButtonRef,
+      previousButtonRef,
+    });
+  });
```

Comment:
should this happen on every change?

Reply:
Since this call will set update the `hidden` boolean to hide/show our arrows, yes. Did you meant to add the concerned refs to the dependencies?